tick/optim/model/src/logreg.cpp

```diff
@@ -45,39 +45,38 @@ double ModelLogReg::grad_i_factor(const ulong i, const ArrayDouble &coeffs) {
 }
 
 double ModelLogReg::sdca_dual_min_i(const ulong i,
-                                    const ArrayDouble &dual_vector,
+                                    const double dual_i,
                                     const ArrayDouble &primal_vector,
-                                    const ArrayDouble &previous_delta_dual,
-                                    const double l_l2sq) {
+                                    const double previous_delta_dual_i,
+                                    double l_l2sq) {
   compute_features_norm_sq();
   double epsilon = 1e-1;
   double normalized_features_norm = features_norm_sq[i] / (l_l2sq * n_samples);
   if (use_intercept()) {
     normalized_features_norm += 1. / (l_l2sq * n_samples);
   }
   const double primal_dot_features = get_inner_prod(i, primal_vector);
-  const double dual = dual_vector[i];
   const double label = get_label(i);
   double new_dual_times_label{0.};
 
   // initial delta dual as suggested in original paper
   // http://www.jmlr.org/papers/volume14/shalev-shwartz13a/shalev-shwartz13a.pdf 6.2
-  double delta_dual = label / (1. + exp(primal_dot_features * label)) - dual;
+  double delta_dual = label / (1. + exp(primal_dot_features * label)) - dual_i;
```

Comment:
This is legacy code as well. This is not a very sensitive line though as it simply initializes the initial Newton descent point.